shell.nix

```diff
@@ -0,0 +1,46 @@
+let
+  here = toString ./.;
+  moz_overlay = import (builtins.fetchTarball
+    "https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz");
+  pkgs = import <nixpkgs> { overlays = [ moz_overlay ]; };
+  rust = (pkgs.rustChannelOf {
+    channel = "nightly";
```

Comment:
Can you use `rustToolchain = "crates/rust-toolchain.toml"` so this follows the supported nightly toolchain?